comp/logs/agent/agentimpl/agent.go

```diff
@@ -240,9 +237,31 @@ func (a *logAgent) startPipeline() {
 		a.pipelineProvider,
 		a.diagnosticMessageReceiver,
 		a.launchers,
-		a.schedulers,
 	)
 	starter.Start()
+
+	if !sds.ShouldBlockCollectionUntilSDSConfiguration(a.config) {
+		a.startSchedulers()
+		a.log.Info("logs-agent started")
+
+		a.started.Store(status.StatusRunning)
```

Comment:
This seems duplicate with line 263.

Reply:
It's fine either inside or outside of `startSchedulers`, but there should be one store on each code path. Would it be possible to remove the store from `startSchedulers`?